src/Raven.Server/Rachis/RachisLogHistory.cs

```diff
@@ -177,6 +177,37 @@ public void UpdateHistoryLog(ClusterOperationContext context, long index, long t
             UpdateInternal(context, cmd, guid, type, index, term, HistoryStatus.Committed, result, exception);
         }
 
+        public void UpdateHistoryLogPreservingGuidAndStatus(ClusterOperationContext context, long index, long term, BlittableJsonReaderObject cmd, object result, Exception exception)
+        {
+            var list = GetLogByIndex(context, index);
+            if (list == null || list.Count != 1)
+            {
+                return;
+            }
+
+            var cmdDjv = list[0];
+
+            var guid = (string)cmdDjv["Guid"];
+            if (guid == null)
+                return;
```

Comment:
I assume that this is why you removed the unique raft id, but what is the purpose for that?